### PR TITLE
Check that type names are resolved

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

RuntimeStdlibTest now checks that names in type signatures are qualified (aka poor man's typechecker)

Will be merged after the stdlib tidying by James is complete.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] ~If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.~

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso [hunk 1]
```diff
@@ -370,7 +371,7 @@ type Date
              from Standard.Base import Date, Time_Of_Day, Time_Zone
 
              example_to_time = Date.new 2020 2 3 . to_date_time Time_Of_Day.new Time_Zone.utc
-    to_date_time : Time_Of_Day -> Time_Zone -> Date_Time.Date_Time
+    to_date_time : Time_Of_Day -> Time_Zone -> Date_Time_Type
```
Comment (4e6) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:375]: Paths like `Foo.Bar` are not supported in type signatures right now

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
 import project.Error.Common.Panic
+import project.Data.Any.Any
 from project.Data.Boolean import Boolean, False
 from project.Data.Range import all
+import project.Data.Ordering.Ordering
+import project.Data.Text.Text
+import project.Data.Numbers.Integer
+import project.Data.Vector.Vector
 import project.Meta
+import project.Nothing.Nothing
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Array.enso]:
```suggestion
import project.Data.Any.Any
import project.Data.Numbers.Integer
import project.Data.Ordering.Ordering
import project.Data.Text.Text
import project.Data.Vector.Vector
import project.Error.Common.Panic
import project.Meta
import project.Nothing.Nothing

from project.Data.Boolean import Boolean, False
from project.Data.Range import all
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso [hunk 3]
```diff
@@ -13,6 +14,7 @@ from project.Data.Vector import Empty_Error
 from project.Error.Common import Error, Panic, Illegal_Argument_Error_Data, Illegal_Argument_Error, Unsupported_Argument_Types, Unsupported_Argument_Types_Data, Incomparable_Values_Error
 
 polyglot java import org.enso.base.statistics.Moments
+polyglot java import org.enso.base.statistics.MomentStatistic
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Statistics.enso:17]: Will need to add this to the `hiding` list in Main.enso.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso [hunk 4]
```diff
@@ -1232,7 +1232,7 @@ Text.locate self term="" mode=Matching_Mode.First matcher=Text_Matcher.Case_Sens
          match_1 . map .length == [2, 3]
          match_2 = ligatures . locate_all "ffiff" matcher=(Text_Matcher Case_Insensitive)
          match_2 . map .length == [2, 5]
-Text.locate_all : Text -> Matcher -> [Span]
+Text.locate_all : Text -> (Text_Matcher | Regex_Matcher) -> Vector
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso]:
```suggestion
Text.locate_all : Text -> (Text_Matcher | Regex_Matcher) -> Vector Span
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso (hunk 1 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:375]: Can we add a to do to this (and the Regex Engine one) so we track these two.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso [hunk 5]
```diff
@@ -10,52 +10,62 @@ type Function
...
 
 ## Converts a single-argument function accepting a pair of elements into a multi-argument one.
 
+   curry : ([a, b] -> c) -> (a -> b -> c)
+
    Arguments:
    - f function accepting pair of values
 
-curry : ([a, b] -> c) -> (a -> b -> c)
+curry : (Vector -> Any) -> (Any -> Any -> Any)
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso]:
```suggestion
curry : (Vector Any -> Any) -> (Any -> Any -> Any)
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso [hunk 6]
```diff
@@ -10,52 +10,62 @@ type Function
...
 
 ## Converts a multi-argument function into a single-argument one accepting a pair of elements.
 
+   uncurry : (a -> b -> c) -> ([a, b] -> c)
+
    Arguments:
    - f function accepting multiple arguments
 
-uncurry : (a -> b -> c) -> ([a, b] -> c)
+uncurry : (Any -> Any -> Any) -> (Vector -> Any)
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Function.enso]:
```suggestion
uncurry : (Any -> Any -> Any) -> (Vector Any -> Any)
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Nothing.enso [hunk 7]
```diff
@@ -1,5 +1,4 @@
-import project.Data.Any.Any
-from project.Data.Boolean import Boolean, True
+from Standard.Base import Any, Boolean, True
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Nothing.enso]:
```suggestion
import project.Data.Any.Any
from project.Data.Boolean import Boolean, True
```
Please keep this.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Runtime.enso [hunk 8]
```diff
@@ -19,7 +22,7 @@ primitive_get_stack_trace = @Builtin_Method "Runtime.primitive_get_stack_trace"
 
    Returns the execution stack trace of its call site. The ordering of the
    resulting vector is such that the top stack frame is the first element.
-get_stack_trace : Vector Stack_Trace_Element
+get_stack_trace : Vector_Type
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Runtime.enso]:
```suggestion
get_stack_trace : Vector Stack_Trace_Element
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Runtime/Unsafe.enso [hunk 9]
```diff
@@ -2,6 +2,10 @@
    A container for unsafe operations that operate based on implementation
    details of the language.
 
+import project.Data.Any.Any
+import project.Data.Numbers.Integer
+from Standard.Base.Meta import Atom
```
Comment (jdunkerley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Runtime/Unsafe.enso]:
```suggestion
import Standard.Base.Meta.Atom
```

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso [hunk 10]
```diff
@@ -28,6 +29,7 @@ import project.Internal.IR.Query.Query
 
 from Standard.Database.Errors import Unsupported_Database_Operation_Error_Data
 
+polyglot java import java.sql.Connection
```
Comment (jdunkerley) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso]: Should be:
```suggestion
import project.Connection.Connection.Connection
```

File: distribution/lib/Standard/Google_Api/0.0.0-dev/src/Main.enso [hunk 11]
```diff
@@ -49,12 +49,11 @@ type Spreadsheets
    - secret_file: a file containing Google Service Account credentials to use to
      access Google services. The credentials file can be downloaded from the
      Google Admin Console when generating a key.
-initialize : File.File -> Google_Api_Client
+initialize : File -> Google_Api_Client
```
Comment (jdunkerley) [inline distribution/lib/Standard/Google_Api/0.0.0-dev/src/Main.enso]: Think should be `File.File` still so guess need
```
import Standard.Base.System.File.File as File_Type
```

Not a biggie as we don't use this library currently.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Reader.enso [hunk 12]
```diff
@@ -1,5 +1,6 @@
 from Standard.Base import all
 import Standard.Base.Error.Common as Errors
+from Standard.Base.System.File import Input_Stream
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Reader.enso]:
```suggestion
import Standard.Base.System.File.Input_Stream
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso (hunk 1 above)
Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:375]: Are all kinds of paths forbidden? Or is it just that _constructors_ cannot be treated as types?

I don't understand why the former syntactic restriction on paths would be placed? If I have a type in module `Standard.Base.Foo.Bar.Baz` and I import `Foo`, I don't see why I cannot refer to it by `Foo.Bar.Baz`? (As long as it is still a _type_ and not a _constructor_, of course)

File: distribution/lib/Standard/Table/0.0.0-dev/src/Excel/Excel_Reader.enso [hunk 13]
```diff
@@ -27,7 +27,7 @@ prepare_reader_table on_problems result_with_problems =
 
 ## PRIVATE
    Convert Boolean|Infer to the correct HeaderBehavior
-make_java_headers : (Boolean|Infer) -> ExcelHeaders.HeaderBehavior
+make_java_headers : (Boolean | Infer) -> Any
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Excel/Excel_Reader.enso]: Lets just lose this line.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso [hunk 14]
```diff
@@ -10,6 +10,7 @@ import project.Internal.Table_Helpers
 import project.Internal.Unique_Name_Strategy.Unique_Name_Strategy
 
 from project.Data.Aggregate_Column.Aggregate_Column import all
+from project.Data.Table import Table
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso]:
```suggestion
import project.Data.Table.Table
```

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso [hunk 15]
```diff
@@ -45,7 +46,7 @@ type Validated_Aggregate_Columns
    - resolves the column descriptors, reporting any issues,
    - ensures that the output names are unique,
    - finds the key columns.
-prepare_aggregate_columns : [Aggregate_Column] -> Table -> Problem_Behavior -> Resolved_Aggregate_Columns
+prepare_aggregate_columns : Vector -> Table -> Problem_Behavior -> Validated_Aggregate_Columns
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Aggregate_Column_Helper.enso]:
```suggestion
prepare_aggregate_columns : Vector Aggregate_Column -> Table -> Problem_Behavior -> Validated_Aggregate_Columns
```

**Review by jdunkerley — APPROVED** [review]
A few minor corrections.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso (hunk 1 above)
Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:375]: Thought Dmitry, was just saying we need to alias types for now as doesn't support path style currently. I'm assuming this is just a checker thing and hopefully a small issue for us for time being.

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:375]: I see, I think I'm wondering if it is a temporary limitation or just that we don't plan to support it at all.

If former then sounds all totally OK, if latter I'd like to reconsider.

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:375]: Yes should only be temporary.

Comment (4e6) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso:375]: >  If I have a type in module Standard.Base.Foo.Bar.Baz and I import Foo, I don't see why I cannot refer to it by Foo.Bar.Baz?

@radeusgd it's a bug. I didn't have time to look into it. I'll create an issue and put a `TODO` note next to every such workaround.

**Review by JaroslavTulach — APPROVED** [review]

**Pull request merged**